src/lib.rs

```diff
@@ -681,6 +700,198 @@ impl<'a> Default for LocalExecutor<'a> {
+unsafe impl Sync for LeakedExecutor {}
+
+#[cfg(feature = "leak")]
+impl UnwindSafe for LeakedExecutor {}
+#[cfg(feature = "leak")]
+impl RefUnwindSafe for LeakedExecutor {}
+
+#[cfg(feature = "leak")]
+impl fmt::Debug for LeakedExecutor {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        debug_state(self.state, "Executor", f)
+    }
+}
+
+#[cfg(feature = "leak")]
+impl LeakedExecutor {
+    /// Spawns a task onto the executor.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use async_executor::Executor;
+    ///
+    /// let ex = Executor::new().leak();
+    ///
+    /// let task = ex.spawn(async {
+    ///     println!("Hello world");
+    /// });
+    /// ```
+    pub fn spawn<T: Send + 'static>(
+        &self,
+        future: impl Future<Output = T> + Send + 'static,
+    ) -> Task<T> {
+        let (runnable, task) = Builder::new()
+            .propagate_panic(true)
+            .spawn(|()| future, self.schedule());
+        runnable.schedule();
+        task
+    }
+
+    /// Spawns a non-`'static` task onto the executor.
+    ///
+    /// ## Safety
+    /// The caller must ensure that the returned task terminates
+    /// or is cancelled before the end of 'a.
+    pub unsafe fn spawn_scoped<'a, T: Send + 'static>(
+        &self,
+        future: impl Future<Output = T> + Send + 'a,
+    ) -> Task<T> {
+        // SAFETY:
+        //
+        // - `future` is `Send`
+        // - `future` is not `'static`, but the caller guarantees that the
+        //    task, and thus its `Runnable` must not live longer than `'a`.
+        // - `self.schedule()` is `Send`, `Sync` and `'static`, as checked below.
+        //    Therefore we do not need to worry about what is done with the
+        //    `Waker`.
+        let (runnable, task) = unsafe {
+            Builder::new()
+                .propagate_panic(true)
+                .spawn_unchecked(|()| future, self.schedule())
+        };
+        runnable.schedule();
+        task
+    }
+
+    /// Attempts to run a task if at least one is scheduled.
+    ///
+    /// Running a scheduled task means simply polling its future once.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use async_executor::Executor;
+    ///
+    /// let ex = Executor::new().leak();
+    /// assert!(!ex.try_tick()); // no tasks to run
+    ///
+    /// let task = ex.spawn(async {
+    ///     println!("Hello world");
+    /// });
+    /// assert!(ex.try_tick()); // a task was found
+    /// ```
+    pub fn try_tick(&self) -> bool {
+        match self.state.queue.pop() {
+            Err(_) => false,
+            Ok(runnable) => {
+                // Notify another ticker now to pick up where this ticker left off, just in case
+                // running the task takes a long time.
+                self.state.notify();
+
+                // Run the task.
+                runnable.run();
+                true
+            }
+        }
+    }
```

Comment:
This shouldn't be duplicated from `Executor::try_tick`. It should be moved to `State` if necessary.

Reply:
Ditto for the other functions.

Reply:
Done.